src/Components/Server/src/ComponentHub.cs

```diff
@@ -122,26 +134,35 @@ public string StartCircuit(string baseUri, string uri)
                 return null;
             }
 
-            var circuitHost = _circuitFactory.CreateCircuitHost(
-                Context.GetHttpContext(),
-                circuitClient,
-                baseUri,
-                uri,
-                Context.User);
-
-            circuitHost.UnhandledException += CircuitHost_UnhandledException;
-
-            // Fire-and-forget the initialization process, because we can't block the
-            // SignalR message loop (we'd get a deadlock if any of the initialization
-            // logic relied on receiving a subsequent message from SignalR), and it will
-            // take care of its own errors anyway.
-            _ = circuitHost.InitializeAsync(Context.ConnectionAborted);
-
-            _circuitRegistry.Register(circuitHost);
-
-            CircuitHost = circuitHost;
-
-            return circuitHost.CircuitId;
+            try
+            {
+                var circuitHost = _circuitFactory.CreateCircuitHost(
+                    Context.GetHttpContext(),
+                    circuitClient,
+                    baseUri,
+                    uri,
+                    Context.User);
+
+                circuitHost.UnhandledException += CircuitHost_UnhandledException;
+
+                // Fire-and-forget the initialization process, because we can't block the
+                // SignalR message loop (we'd get a deadlock if any of the initialization
+                // logic relied on receiving a subsequent message from SignalR), and it will
+                // take care of its own errors anyway.
+                _ = circuitHost.InitializeAsync(Context.ConnectionAborted);
+
+                // It's safe to *publish* the circuit now because nothing will be able
+                // to run inside it until after InitializeAsync completes.
+                _circuitRegistry.Register(circuitHost);
+                CircuitHost = circuitHost;
+                return circuitHost.CircuitId;
+            }
+            catch (Exception ex)
+            {
+                Log.CircuitInitializationFailed(_logger, ex);
+                NotifyClientError(Clients.Caller, "The circuit failed to initialize.");
```

Comment:
nit:
```suggestion
                _ = NotifyClientError(Clients.Caller, "The circuit failed to initialize.");
```

Reply:
Same comment as above

Reply:
same response as above - I have another work item to fix this by terminating the connection.